### ISSUE TITLE
Group block: reconsider the Orientation setting

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#62740]
### Description

I'm not sure I fully understand what the purpose of the 'Orientation' setting in the Group block is.

I may be missing something but it seems to me that it duplicates sxisting functionalities and controls.

The Group block has now 4 variants:
- Group
- Row
- Stack
- Grid 

The Orientation setting is available only for the Row and Stack variants. To my understanding, it just switches the block from Row to Stack and vice-versa. There's already an UI for that. As such, the Orientation settings appears to be redundant. It only adds cognitive load and it's potentially confusing for users.

Animated GIF to illustrate:

![orientation](https://github.com/WordPress/gutenberg/assets/1682452/4c9a8f82-7e4c-4672-a6fc-e46ccc8366d7)

### Step-by-step reproduction instructions

- Add a Group block and add a couple images or a couple paragraphs within it.
- In the Inspetor, change the Group to be a Row.
- Observe the Orientation setting appears in the Inspector.
- Switch the orientation and observe it just switches the Row to Stack.
- Switch the orientation again and observe it jut switches the Stack back to a Row.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by mtias** [comment]
Variants map over attributes, so they can be seen as shortcuts for certain attribute combinations. There's been a few explorations to reduce, but I think so far it's more instructive than redundant: Row and Stack as flex containers that have different orientations, they can teach how containers work without obfuscating.

**Comment by afercia** [comment]
I understand the reasoning but I'm not sure this UI helps users.
Also, I'd argue the UI is conceptually wrong.

No matter what, these variants are presented to users as 'block types'.
- They're listed separately in the Inserter as they were different blocks.
- When switching variant, the Block inspector card shows the variant name as it was a block. It doesn't say 'Group, stack cariant'. It does say 'Stack'.

Regardless of the technical implementation, the UI presents these variants as blocks. As a user, I understand they are 4 different type of blocks.

The block settings panel (the inspector) is supposed to show settings for the selected block. 
Instead, the Orientation 'setting' isn't a setting for the Row or the Stack. It's just a switcher. It doesn't change an attribute of the Stack or Row (seen by users as blocks). It changes the variant. To me, the inspector should only contain settings of the selected block (even when it's technically a variant).

**Comment by mtias** [comment]
What variants aim to express is that certain blocks that you can individually find on the inserter are still related and part of the same family. That may be a subtle but important distinction in helping people understand a generally complex thing like layout primitives. Not saying the UI is quite there yet, there's room for improvement, but it's trying to balance discoverability (be easy to add from the inserter when you want to express a specific container type), clarity (if you insert a pattern, to be able to quickly understand how it's built by seeing and reading its block type), and interoperability (they are all the same block underneath, which makes the available tools consistent). I can see hiding the orientation tool, but I don't know that it improves things, if anything it makes "row" and "stack" seem more opaque regarding what they are doing.

**Comment by afercia** [comment]
I'd tend to think one of the root issues here is the way variants are communicated in the UI.

As a user, I wouldn't understand they are variants of the `Group` block.
Actually, the UI presentes the variants as if they were different block types.
For examxple, the name in the Inspector block card changes and doesn't mention `Group`. it only says `Row`, `Stack`, or `Grid`.
It's even more disorienting when, as a user, I inseert a variant from the inserter of from the slash inserter.
I may search in the inserter and add directly a `Stack`. In this case, the UI doesn't inform me anywhere that this is actually a variant of the Group block.
